images/virtualization-artifact/pkg/controller/service/disk_service.go

```diff
@@ -0,0 +1,130 @@
+package service
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	cdiv1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/deckhouse/virtualization-controller/pkg/controller/kvbuilder"
+	"github.com/deckhouse/virtualization-controller/pkg/controller/supplements"
+	"github.com/deckhouse/virtualization-controller/pkg/dvcr"
+	"github.com/deckhouse/virtualization-controller/pkg/sdk/framework/helper"
+	virtv2 "github.com/deckhouse/virtualization/api/core/v1alpha2"
+)
+
+type DiskService struct {
+	client       client.Client
+	clientset    kubernetes.Interface
+	dvcrSettings *dvcr.Settings
+}
+
+func NewDiskService(
+	client client.Client,
+	clientset kubernetes.Interface,
+	dvcrSettings *dvcr.Settings,
+) *DiskService {
+	return &DiskService{
+		client:       client,
+		clientset:    clientset,
+		dvcrSettings: dvcrSettings,
+	}
+}
+
+func (s DiskService) Start(
+	ctx context.Context,
+	pvcSize resource.Quantity,
```

Comment:
Return pointer type from constructor but define methods for value type is misleading. The first intention is to ask why is that? =)

Reply:
The constructor returns a pointer to make it convenient to substitute services under interfaces. The methods do not have a pointer because the methods do not change the structure of the service.